src/org/jitsi/impl/neomedia/rtcp/RTCPTermination.java

```diff
@@ -0,0 +1,565 @@
+            // Also, the cache MUST be notified of packets being retransmitted,
+            // in order for it to update their timestamp. We do this here by
+            // simply letting retransmitted packets pass through the cache again.
+            // We use the retransmission requester here simply because it is
+            // the transformer right before the cache, not because of anything
+            // intrinsic to it.
+            RetransmissionRequester rr = stream.getRetransmissionRequester();
+            TransformEngine after
+                = (rr instanceof TransformEngine) ? (TransformEngine) rr : null;
+
+            long rtt = stream.getMediaStreamStats().getSendStats().getRtt();
+            long now = System.currentTimeMillis();
+
+            for (Iterator<Integer> i = lostPackets.iterator(); i.hasNext();)
+            {
+                int seq = i.next();
+                RawPacketCache.Container container
+                    = cache.getContainer(ssrc, seq);
+
+
+                if (container != null)
+                {
+                    // Cache hit.
+                    long delay = now - container.timeAdded;
+                    boolean send = (rtt == -1) ||
+                        (delay >= Math.min(rtt * 0.9, rtt - 5));
+
+                    if (logger.isDebugEnabled())
+                    {
+                        logger.debug(Logger.Category.STATISTICS,
+                            "retransmitting,stream=" + stream.hashCode()
+                                + " ssrc=" + ssrc
+                                + ",seq=" + seq
+                                + ",send=" + send);
+                    }
+
+                    if (send && rtxTransformer.retransmit(container.pkt, after))
+                    {
+                        statistics.packetsRetransmitted.incrementAndGet();
+                        statistics.bytesRetransmitted.addAndGet(
+                            container.pkt.getLength());
+                        i.remove();
+                    }
+
+                    if (!send)
+                    {
+                        statistics.packetsNotRetransmitted.incrementAndGet();
+                        statistics.bytesNotRetransmitted.addAndGet(
+                            container.pkt.getLength());
+                        i.remove();
+                    }
+
+                }
+                else
+                {
+                    statistics.packetsMissingFromCache.incrementAndGet();
+                }
+            }
+        }
+
+        if (!lostPackets.isEmpty())
+        {
+            // If retransmission requests are enabled, videobridge assumes
+            // the responsibility of requesting missing packets.
+            logger.debug("Packets missing from the cache.");
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public long getTimeUntilNextRun()
+    {
+        return (lastRunMs + REPORT_PERIOD_MS) - System.currentTimeMillis();
```

Comment:
Math.max(0, ...)?

I think it would be better to inherit from PeriodicRunnable and implement RTCPPacketListener (it only has empty methods to be added, while the PeriodicRunnable has some logic which has to be duplicated)